lib/elliptic/ec/index.js

```diff
@@ -164,3 +164,44 @@ EC.prototype.verify = function verify(msg, signature, key, enc) {
 
   return p.getX().mod(this.n).cmp(r) === 0;
 };
+
+EC.prototype.recoverPubKey = function(msg, signature, j) {
+  assert(3 & j === j, 'The recovery param is more than two bits');
+  var n = this.n;
+  var e = new bn(msg);
+  var r = signature.r;
+  var s = signature.s;
+
+  // A set LSB signifies that the y-coordinate is odd
+  var isYOdd = j & 1;
+
+  // The more significant bit specifies whether we should use the
+  // first or second candidate key.
+  var isSecondKey = j >> 1;
+
+  // 1.1. Let x = r + jn.
+  var x = isSecondKey ? n.add(r) : r;
```

Comment:
You may skip this check altogether and throw if `r` is bigger than that value and `isSecondKey` is `true`.